src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -417,6 +417,12 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     responseStream = EmptyReadStream.Instance;
                     ReturnConnectionToPool();
                 }
+                else if (request.Method == s_httpConnectMethod && (int)response.StatusCode == 200)
```

Comment:
It doesn't seem like there's any logic here to format the CONNECT target properly, see https://tools.ietf.org/html/rfc7231#section-4.3.6.

The request should look something like "CONNECT foo.com:345 HTTP/1.1".

The value for the target host should come from the Host header (and if not present, we should presumably throw).

Reply:
I had some changes originally for the proxy support but I left them outside of this PR. 
I'll fix it.